aspnetcore/security/enforcing-ssl.md

```diff
@@ -41,38 +43,43 @@ The following code calls `UseHttpsRedirection` in the `Startup` class:
 
 The preceding highlighted code:
 
-* Uses the default [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) (`Status307TemporaryRedirect`).
+* Uses the default [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) ([Status307TemporaryRedirect](/dotnet/api/microsoft.aspnetcore.http.statuscodes.status307temporaryredirect)).
 * Uses the default [HttpsRedirectionOptions.HttpsPort](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.httpsport) (null) unless overridden by the `ASPNETCORE_HTTPS_PORT` environment variable or [IServerAddressesFeature](/dotnet/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature).
 
-> [!WARNING] 
->A port must be available for the middleware to redirect to HTTPS. If no port is available, redirection to HTTPS does not occur. The HTTPS port can be specified by any of the following setting:
-> 
->* `HttpsRedirectionOptions.HttpsPort` 
->* The `ASPNETCORE_HTTPS_PORT` environment variable. 
->* In development, an HTTPS url in *launchsettings.json*. 
->* An HTTPS url configured directly on Kestrel or HttpSys. 
+We recommend using temporary redirects rather than permanent redirects, as link caching can cause unstable behavior in development environments. We recommend using [HSTS](#hsts) to signal to clients that only secure resource requests should be sent to the app (only in production).
+
+> [!WARNING]
+> A port must be available for the middleware to redirect to HTTPS. If no port is available, redirection to HTTPS doesn't occur. The HTTPS port can be specified using any of the following approaches:
+>
+> * Set `HttpsRedirectionOptions.HttpsPort`.
+> * Set the `ASPNETCORE_HTTPS_PORT` environment variable.
+> * In development, set an HTTPS URL in *launchsettings.json*.
+> * Configure an HTTPS URL endpoint for [Kestrel](xref:fundamentals/servers/kestrel) or [HTTP.sys](xref:fundamentals/servers/httpsys).
+>
+> When Kestrel or HTTP.sys is used as an edge server, Kestrel or HTTP.sys must be configured to listen on both the secure port where the client is redirected (typically, 443 in production and 5001 in development) and the insecure port (typically, 80 in production and 5000 in development). The insecure port must be accessible by the client in order for the app to receive an insecure request and redirect it to the secure port. Any firewall between the client and server must also have the ports open for traffic. For more information, see [Kestrel endpoint configuration](xref:fundamentals/servers/kestrel#endpoint-configuration) or <xref:fundamentals/servers/httpsys>.
```

Comment:
- edge server --> public-facing edge server
- Split up the first sentence. Maybe use a couple bullets after "must be configured to listen on".

Reply:
> public-facing edge server

We haven't been doing that in general. I think we generally rely on a dev Googling it if they don't know what it means. I feel that that new devs will benefit from the adjective. I've opened https://github.com/aspnet/Docs/issues/9000 to take a look.